src/server/api/routers/user.ts

```diff
@@ -203,6 +203,332 @@ export const userRouter = createTRPCRouter({
+            return await ctx.prisma.user.findMany({
+                where: {
+                    accountStatus: 'Unlinked',
+                },
+                select: {
+                    id: true,
+                    discordUsername: true,
+                    discordDiscriminator: true,
+                },
+                orderBy: getOrderObjectArray(input.selectors, input.directions),
+            });
+        }),
+
+    getUserClaims: loggedInProcedure
+        .query(async ({ ctx }) => {
+            return await ctx.prisma.userClaim.findMany({
+                where: {
+                    claimByUserId: ctx.user.id,
+                },
+                select: {
+                    id: true,
+                    claimForUserId: true,
+                    User_UserClaim_claimFor: {
+                        select: {
+                            discordUsername: true,
+                            discordDiscriminator: true,
+                        }
+                    }
+                },
+                orderBy: {
+                    id: 'asc'
+                },
+            });
+        }),
+    
+    getAllUserClaims: adminProcedure
+        .query(async ({ ctx }) => {
+            return await ctx.prisma.userClaim.findMany({
+                select: {
+                    id: true,
+                    User_UserClaim_claimBy: {
+                        select: {
+                            discordUsername: true,
+                            discordDiscriminator: true,
+                        }
+                    },
+                    User_UserClaim_claimFor: {
+                        select: {
+                            discordUsername: true,
+                            discordDiscriminator: true,
+                        }
+                    },
+                },
+                orderBy: {
+                    id: 'asc',
+                }
+            });
+        }),
+
+    createUserClaim: loggedInProcedure
+        .input(z.object({
+            forUserId: z.string()
+        }))
+        .mutation(async ({ ctx, input }) => {
+            const claim = await ctx.prisma.userClaim.create({
+                data: {
+                    claimByUserId: ctx.user.id,
+                    claimForUserId: input.forUserId
+                },
+            });
+
+            return claim;
+        }),
+    
+    verifyUserClaim: adminProcedure
+        .input(z.object({
+            id: z.number()
+        }))
+        .mutation(async ({ ctx, input }) => {
+            const claim = await ctx.prisma.userClaim.findFirst({
+                where: {
+                    id: input.id,
+                }
+            });
+            if (!claim) {
+                throw new TRPCError({
+                    code: "NOT_FOUND",
+                    message: `No user claim exists with id "${input.id}"`,
+                });
+            }
+
+            await ctx.prisma.publisher.updateMany({
```

Comment:
Any suggestions on improving these 200 lines of foreign key changes? (Some of the inner objects are common so I guess I could create them once and then use them.)

Reply:
Yeah I added those common objects, added a check that the users in the claim actually exist, and wrapped it all in a transaction. The reason for the check that the users still exist, is so that if the db is ever in a slightly unexpected state, we don't make it worse by silently adding references to non-extant foreign keys. In most of the tRPC api (not sure if the oldest parts were done this way... i should check lol), we add foreign keys with the `{ connect: {...}}` syntax which automatically makes sure that the foreign keys are, in fact, valid. We can't use that here as `updateMany` doesn't support it, so we perform the check in our application instead and wrap it all in an [interactive transaction](https://www.prisma.io/docs/orm/prisma-client/queries/transactions#interactive-transactions).